### ISSUE TITLE
HMR not working

### ISSUE THREAD

**Issue by p0d4r14n** [elysiajs/elysia#4]
Hi, it seems like HMR is not working. Is this feature supposed to work? I expected so because the dev script includes `--hot`.

How to reproduce:
1. Create a new Elysia project: `bun create elysia elysia-playground`
2. Start the dev server with --hot: `bun run dev` which will execute `bun run --hot src/index.ts`
3. Change something in the code, i.e. **Hello Elysia** to **Hello** in _src/index.ts_
4. It seems like the change is noticed as the console output states but there is no page refresh and manual triggered page refresh still only shows **Hello Elysia**

```
$ bun run --hot src/index.ts
🦊 Elysia is running at 0.0.0.0:3000
    Dir change: src/
    Dir change: src
    Dir change: src
    Dir change: src
    Dir change: src
```

bun version: 0.4.0

```
elysia-playground (master *) % bun pm ls
elysia-playground node_modules (2)
├── bun-types@0.4.0
└── elysia@0.1.0
```

OS: Mac Ventura 13.0
Chrome Version 107.0.5304.87

**Comment by SaltyAom** [comment]
Ok, I can confirm that 0.1.0 doesn't work on my side too, the issue should be fixed in 0.1.1, try updating to see if the issue still persists.

Also unlike the client-side library, the web framework usually required you to do a page refresh because we can't inject JavaScript into the page to make it reload.

Let me know if this help 😄

**Comment by p0d4r14n** [comment]
Sadly, doesn't seem to work with 0.1.1 either. I just updated the version in previously generated project, mentioned above.
```
elysia-playground node_modules (2)
├── bun-types@0.4.0
└── elysia@0.1.1
```

**Comment by SaltyAom** [comment]
I'm not sure if Bun cached the installation or not, can you try to remove `node_modules` and `bun.lockb` and reinstall to see if it was working?

If all doesn't work, then can you try this [repo](https://github.com/SaltyAom/elysia-hmr)?

Here running at MacOS Ventura 13.0.1.

https://user-images.githubusercontent.com/35027979/209837430-eb2c5bd0-e362-4813-b3ad-1028fb1119a8.mp4

If it still doesn't work then maybe it's a browser cache, or a pending process that doesn't get killed, try restarting the machine to see if it works.

**Comment by bobwatcherx** [comment]
maybe this helps
https://stackoverflow.com/questions/73208846/hot-reload-hmr-with-bun-dev

**Comment by p0d4r14n** [comment]
Can't get it to work, also tried a fresh project. Same problem with the repo @SaltyAom linked. Tried deleting cache and different browsers and restarting my machine.
What version of bun are you running?

**Comment by bobwatcherx** [comment]
I am use buj newest version

I don't use command
Bun run --hot src/main.ts

But use nodemon , and this worked

Check link from me 
Check answer 1

**Comment by zafranf** [comment]
hi, i just try elysia today..
when fresh install i got `elysia@0.0.0-experimental.56`, then upgrade to `0.1.1` still no luck, then i use `bun add elysia@latest` i got `0.1.2` and its working now..

**Comment by mateo-m** [comment]
Just tried using Elysia 0.1.2, but it didn't fix this issue on my end.

**Comment by SaltyAom** [comment]
Does anyone use WSL by any chance? The problem seems to occur mostly WSL.

For anyone who is using WSL, can you try the code below, if it still doesn't reload, then it's probably on Bun:
```
export default {
    port: 8080,
    fetch: (request) => new Response('hi')
}
```

Please add a thumb up 👍, if this happens to you on WSL, so we can report the issue to Bun.

You can use nodemon for a workaround.

**Comment by mateo-m** [comment]
Could this https://github.com/Jarred-Sumner/bun-livereload be added to Elysia as a band-aid until a fix for Bun is released?

**Comment by An-Huynh-Certivity** [comment]
I am on Debian and have observed the following behavior:

When initiating the project with `bun create elysia server` it downloads the following versions:
```
❯ bun pm ls  
/home/an/coding/elysia-test/server node_modules (2)
├── bun-types@0.2.2
└── elysia@0.0.0-experimental.56
```

After deleting `node_modules` and `bun.lockb`, and running `bun install`, I now have the following versions downloaded:
```
/home/an/coding/elysia-test/server node_modules (2)
├── bun-types@0.4.0
└── elysia@0.1.2
```

In both cases, HMR is not working at all for me. In fact, it doesn't even detect any changes when saving changes, unlike the original poster. 

I have attached a video demonstrating my issue:

https://user-images.githubusercontent.com/100586735/211034606-817b36ad-3beb-4696-b8e5-481e973c248c.mp4

**Comment by mateo-m** [comment]
The issue seems to be fixed in the latest canary version at the time I'm writing this message.

Source : https://discord.com/channels/876711213126520882/1059617902204031066/1068516159097409646

**Comment by SaltyAom** [comment]
This issue should be fixed in Bun 0.5.3.
